server/src/bots/telegram/services/messageRegistry.ts

```diff
@@ -56,4 +64,36 @@ class MessageRegistry {
 
 export const registry = new MessageRegistry();
 
-type CallBackQueryHandler = (bot: any, message: any, chatId: unknown) => any
+type CallBackQueryHandler = (bot: any, message: any, chatId: unknown, argument?: any) => any
+
+export function WithCommandArgument(handler: CallBackQueryHandler): any {
+    const context = registry;
+    return (bot: any, message: any, chatId: unknown, data?: string): any => {
+        try{
+            const argument = getArgfromMessage.call(context, data || message.text);
+            handler.call(context, bot, message, chatId, argument)
+        }
+        catch(Error){
+            logger.log('warn', {
+                ...message,
+                warningMessage: Error.message,
+            });
+            bot.sendMessage(chatId, Error.message);
+        }
+    }
+}
+function getArgfromMessage(messageText: string): string{
+    if(!messageText){
+        throw new Error('message could not be empty');
```

Comment:
it will never be empty since `message.text` is there even if it's cb handler for inline keyboard